src/main/java/uk/gov/companieshouse/digitalcertifiedcopyprocessor/consumer/MessageLoggingAspect.java

```diff
@@ -47,11 +47,11 @@ void logAfterMainConsumer(JoinPoint joinPoint) {
     }
 
     private void logMessage(String logMessage, Message<?> incomingMessage) {
-        var topic = Optional.ofNullable((String) incomingMessage.getHeaders().get(KafkaHeaders.RECEIVED_TOPIC))
+        var topic = Optional.of((String) incomingMessage.getHeaders().get(KafkaHeaders.RECEIVED_TOPIC))
```

Comment:
Are we safe to potentially throw an NPE here? Aka is it an invalid scenario for these headers to be null?

Reply:
Optional.ofNullable should handle it if is null without NPE, should we keep that in there?

Reply:
If you have a concrete reason for changing it e.g:
These headers should **never** be null and if they are we want to terminate the application

then the change makes sense, otherwise keeping in the built in null-check from ofNullable means the .orElse() call will provide a default value in the event of a null and execution will continue

Reply:
Agreed, I thought the orElse was handling a null value but looks like the exception is thrown before then

Reply:
Thanks!